test/Elastic.Apm.Tests/DistributedTracing/DistributedTracingTests.cs

```diff
@@ -2,14 +2,56 @@
 // Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
 // See the LICENSE file in the project root for more information
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using Elastic.Apm.DistributedTracing;
+using Elastic.Apm.Libraries.Newtonsoft.Json;
+using Elastic.Apm.Tests.Utilities;
 using FluentAssertions;
 using Xunit;
 
 namespace Elastic.Apm.Tests.DistributedTracing
 {
 	public class DistributedTracingTests
 	{
+		public class TraceStateTestData
+		{
+			public List<string[]> Headers { get; set; }
+			[JsonProperty("is_traceparent_valid")] public bool IsTraceParentValid { get; set; }
+			[JsonProperty("is_tracestate_valid")] public bool? IsTraceStateValid { get; set; }
+		}
+
+		[Theory]
+		[JsonFileData("./TestResources/json-specs/w3c_distributed_tracing.json", typeof(TraceStateTestData))]
+		public void TestCasesFromJsonSpec(TraceStateTestData data)
+		{
+			var traceParents = data.Headers
+				.Where(h => h[0].Equals("traceparent", StringComparison.InvariantCultureIgnoreCase))
+				.Select(h => h?[1]);
```

Comment:
```suggestion
				.Select(h => h[1]);
```

nit: I think the `?` null-checks the array itself (which is never `null`), but not the 1. item. Maybe something like this would make it: ~`.Select(h => h.Length < 2 ? string.Empty : h[1]);`~ Update: `h.ElementAtOrDefault(1)`. 

But I think no null-check is needed here due to the nature of the data from the json file (there is always a 1. item in that array).

Reply:
Nice catch!
I was using `SelectFirstOrDefault` initially (which could produce actual `null` values for the array) before I realized that there were test cases with multiple hits. The `?` is a relict from that.